lib/transformations/loaders/loaders.js

```diff
@@ -3,114 +3,171 @@ const safeTraverse = require('../safeTraverse');
 module.exports = function(j, ast) {
 
 	const createArrayExpression = function(p) {
-		var objs = p.parent.node.value.value.split('!').map(val => j.objectExpression([j.property('init', 
-										j.identifier('loader'), 
-										j.literal(val)
-										)]));
-		var loaderArray = j.arrayExpression(objs);
+		let objs = p.parent.node.value.value.split('!')
```

Comment:
I'm not sure it will work in any context but I agree we should try reuse as much as possible. I'd wait till we see more repetitions before we start adding abstractions.